core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -49,6 +49,15 @@ abstract class TypeToSchema extends TypeUtil.SchemaVisitor<Schema> {
   private static final Schema UUID_SCHEMA =
       LogicalTypes.uuid().addToSchema(Schema.createFixed("uuid_fixed", null, null, 16));
   private static final Schema BINARY_SCHEMA = Schema.create(Schema.Type.BYTES);
+  private static final Schema VARIANT_SCHEMA =
```

Comment:
I think this will break when there are multiple variant columns because Avro requires each record to have a unique name. This is why record names are generated from field IDs. Can you create a test case to validate this?

Reply:
You are right. We need to have different name for each record, otherwise, it will generate only one record.

I generate a new record with the name generated from field ID now. I don't cache it since the type is the same.  

This is the schema with two variant columns:
`{"name":"_v1","type":["null",{"type":"record","name":"r18","fields":[{"name":"metadata","type":"bytes"},{"name":"value","type":"bytes"}]}],"default":null,"field-id":18},{"name":"_v2","type":["null",{"type":"record","name":"r19","fields":[{"name":"metadata","type":"bytes"},{"name":"value","type":"bytes"}]}],"default":null,"field-id":19}]}`